src/lib.rs

```diff
@@ -129,9 +154,28 @@ impl Order {
         self.account.get(&mut self.nonce, challenge_url).await
     }
 
-    /// Get the current state of the order
-    pub async fn state(&mut self) -> Result<OrderState, Error> {
-        self.account.get(&mut self.nonce, &self.order_url).await
+    /// Refresh the current state of the order
+    pub async fn refresh(&mut self) -> Result<&OrderState, Error> {
+        let rsp = self
+            .account
+            .post(None::<&Empty>, self.nonce.take(), &self.url)
+            .await?;
+
+        self.nonce = nonce_from_response(&rsp);
+        self.state = Problem::check::<OrderState>(rsp).await?;
+        Ok(&self.state)
+    }
```

Comment:
Can't we let `certificate` take a timeout value, and then let it try to refresh the order until the order is valid or invalid, or the timeout triggers? It seems like a nicer API. 
The user can trigger another round of trying to get the cert by calling `certificate` again. 
I also found nothing about rate limits for getting the current order state. Let's Encrypt and the ACME spec only mention rate limits for creation of orders.

On the other hand, maybe this would be something for a high-level API (maybe in another create) which hides all the internals (states, etc).

I like the split into `finalize` and `certificate` as it better reflects what is going on spec-wise.

Reply:
I kind of intentionally wanted to keep instant-acme pretty opinionated. Providing a backoff strategy triggered sort of runs counter to that idea, so I'd prefer to avoid that. (Also, in practice I'm not sure it's possible in practice to observe the processing state on Let's Encrypt staging or production infra, so I kinda suspect this is mostly an issue when you run pebble in an underpowered environment?)